auction_keeper/main.py

```diff
@@ -538,40 +564,82 @@ def handle_bid(self, id: int, auction: Auction):
                 return
 
         if bid_price is not None and bid_transact is not None:
+            # Ensure this auction has a gas strategy assigned
```

Comment:
let's move this code in DynamicGasPrice class and have unit tests for this logic as well

Reply:
Not sure I understand; `UpdatableGasPrice` is used when the model is managing price, not the keeper.  I don't think most consumers should use that feature, but I'd prefer not to remove it.

Reply:
my suggestion was to put this in a gas related class (maybe DynamicGasPrice since logic is to figure out the gas price strategy?) instead throwing it all in a main.py god like file

Reply:
The purpose of this method is to handle output from the price model.  This generally means either submitting a new transaction or (attempting to) replace an existing transaction.  This logic has no affinity to `DynamicGasPrice`.

If we want to move this logic out of `main.py`, I'd entertain moving some of it into the `Auction` class (`logic.py`).  To avoid exposing main's `_run_future` method, I'd consider refactoring lines 569-596.  Those lines corner cases where a price model changes behavior:
-  model starts providing a price when it hadn't before
-  model stops providing a price when it had before

Reply:
I pushed a commit with the refactoring described above